Mathlib/MeasureTheory/Measure/DiracProba.lean

```diff
@@ -0,0 +1,196 @@
+    simp only [ContinuousAt] at aux
+    exact aux.comp h
+
+/-- An inverse function to `diracProba` (only really an inverse under hypotheses that
+guarantee injectivity of `diracProba`). -/
+noncomputable def diracProbaInverse : range (diracProba (X := X)) → X :=
+  fun μ' ↦ (mem_range.mp μ'.prop).choose
+
+@[simp] lemma diracProba_diracProbaInverse (μ : range (diracProba (X := X))) :
+    diracProba (diracProbaInverse μ) = μ := (mem_range.mp μ.prop).choose_spec
+
+lemma diracProbaInverse_eq [T0Space X] {x : X} {μ : range (diracProba (X := X))}
+    (h : μ = diracProba x) :
+    diracProbaInverse μ = x := by
+  apply injective_diracProba_of_T0 (X := X)
+  simp only [← h]
+  exact (mem_range.mp μ.prop).choose_spec
+
+/-- In a T0 topological space `X`, the assignment `x ↦ diracProba x` is a bijection to its
+range in `ProbabilityMeasure X`. -/
+noncomputable def diracProbaEquiv [T0Space X] : X ≃ range (diracProba (X := X)) where
+  toFun := fun x ↦ ⟨diracProba x, by exact mem_range_self x⟩
+  invFun := diracProbaInverse
+  left_inv x := by apply diracProbaInverse_eq; rfl
+  right_inv μ := Subtype.ext (by simp only [diracProba_diracProbaInverse])
+
+/-- The composition of `diracProbaEquiv.symm` and `diracProba` is the subtype inclusion. -/
+lemma diracProba_comp_diracProbaEquiv_symm_eq_val [T0Space X] :
+    diracProba ∘ (diracProbaEquiv (X := X)).symm = fun μ ↦ μ.val := by
+  funext μ; simp [diracProbaEquiv]
+
+lemma tendsto_diracProbaEquivSymm_iff_tendsto [T0Space X] [CompletelyRegularSpace X]
+    {μ : range (diracProba (X := X))} (F : Filter (range (diracProba (X := X)))) :
+    Tendsto diracProbaEquiv.symm F (𝓝 (diracProbaEquiv.symm μ)) ↔ Tendsto id F (𝓝 μ) := by
+  have key :=
+    tendsto_diracProba_iff_tendsto (F.map diracProbaEquiv.symm) (x := diracProbaEquiv.symm μ)
+  rw [← (diracProbaEquiv (X := X)).symm_comp_self, ← tendsto_map'_iff] at key
+  simp only [tendsto_map'_iff, map_map, Equiv.self_comp_symm, map_id] at key
+  simp only [← key, diracProba_comp_diracProbaEquiv_symm_eq_val]
+  convert tendsto_subtype_rng.symm
+  exact apply_rangeSplitting (fun x ↦ diracProba x) μ
+
+/-- In a T0 topological space, `diracProbaEquiv` is continuous. -/
+lemma continuous_diracProbaEquiv [T0Space X] :
+    Continuous (diracProbaEquiv (X := X)) :=
+  Continuous.subtype_mk continuous_diracProba mem_range_self
+
+/-- In a completely regular T0 topological space, the inverse of `diracProbaEquiv` is continuous. -/
+lemma continuous_diracProbaEquivSymm [T0Space X] [CompletelyRegularSpace X] :
+    Continuous (diracProbaEquiv (X := X)).symm := by
+  apply continuous_iff_continuousAt.mpr
+  intro μ
+  apply continuousAt_of_tendsto_nhds (y := diracProbaInverse μ)
+  exact (tendsto_diracProbaEquivSymm_iff_tendsto _).mpr fun _ mem_nhd ↦ mem_nhd
+
+/-- In a completely regular T0 topological space `X`, `diracProbaEquiv` is a homeomorphism to
+its image in `ProbabilityMeasure X`. -/
+noncomputable def homeomorph_diracProba [T0Space X] [CompletelyRegularSpace X]
```

Comment:
The name of a def should not contain `_` but should be camelCase instead. I see in the docs that many homeomorphs are called with a name in the namespace, like `Homeomorph.diracProba`. Other options would be `homeomorphDiracProba` or `diracProbaHomeomorph`.

Reply:
Thanks! The last option you suggested seems the most natural to me, so I will go with it now. (The `Homeomorph` namespace does not seem to gain any dot notation in this case, and seems more suitable for general results rather than the very specific homeomorphism here.)